components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/HostObjectUtils.java

```diff
@@ -100,6 +101,32 @@ protected static SubscriberKeyMgtClient getKeyManagementClient() throws APIManag
         }
     }
 
+    /**
+     * Used to get instance of ProviderKeyMgtClient
+     * @return ProviderKeyMgtClient
+     * @throws APIManagementException
+     */
+    protected static ProviderKeyMgtClient getProviderClient() throws APIManagementException {
+        APIManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
+        String url = config.getFirstProperty(APIConstants.API_KEY_MANAGER_URL);
```

Comment:
It looks like "url" is validated at a wrong place. To me, it looks like this particular config attribute validation should ideally have happened when the APIManagerConfiguration is loaded. If there's no particular reason why it was done this way, maybe we could get rid of this unnecessary "if" condition from the getProviderClient method.